rednotebook/util/filesystem.py

```diff
@@ -91,22 +91,18 @@ def __init__(self, config):
         self.forbidden_dirs = [user_home_dir, self.journal_user_dir]
 
     def get_user_dir(self, config):
-        custom = config.read("userDir")
-
-        if custom:
-            # If a custom user dir has been set,
-            # construct the absolute path (if not absolute already)
-            # and use it
-            if not os.path.isabs(custom):
-                custom = os.path.join(self.app_dir, custom)
-            user_dir = custom
-        else:
-            if self.portable:
-                user_dir = os.path.join(self.app_dir, "user")
-            else:
-                user_dir = os.path.join(self.user_home_dir, ".rednotebook")
-
-        return user_dir
+        if not (custom := config.read("userDir")):
```

Comment:
Not everybody knows that custom leaks out of the if-clause.

```suggestion
        custom = config.read("userDir")
        if not custom:
```

Reply:
This is just the intended effect of the walrus operator. Since we are using it on other parts of the codebase, I say its fine to keep it this way.

Reply:
Sounds good.